### PR TITLE
`azurerm_linux_virtual_machine` `azurerm_windows_virtual_machine` - support for `gallery_applications`

### PR DESCRIPTION
**Pull request by myc2h6o**
This is similar to what we have added in VMSS. And `gallery_applications` in VM is updatable
https://github.com/hashicorp/terraform-provider-azurerm/blob/fa35df5cce72dc7466be7c19d0225cc879156127/website/docs/r/linux_virtual_machine_scale_set.html.markdown?plain=1#L363-L373

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result (Failed tests are being fixed as part of #17671)
image

File: website/docs/r/linux_virtual_machine.html.markdown [hunk 1]
```diff
@@ -244,6 +246,20 @@ A `diff_disk_settings` block supports the following:
 
 ---
 
+A `gallery_applications` block supports the following:
+
+* `package_reference_id` - (Required) Specifies the Gallery Application Version resource ID.
+
+-> **NOTE:** The `package_reference_id` should be in the form of `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Compute/galleries/gallery1/applications/application1/versions/version1`.
```
Comment (katbyte) [inline website/docs/r/linux_virtual_machine.html.markdown:253]: should this be called
```suggestion
* `version_id` - (Required) Specifies the Gallery Application Version resource ID.
```
or
```suggestion
* `package_reference_id` - (Required) Specifies the Gallery Application Version resource ID.

-> **NOTE:** The `gallery_application_version_id` should be in the form of `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Compute/galleries/gallery1/applications/application1/versions/version1`.
```

so the note sins't required?

File: website/docs/r/windows_virtual_machine.html.markdown [hunk 2]
```diff
@@ -243,6 +245,20 @@ A `diff_disk_settings` block supports the following:
 
 ---
 
+A `gallery_applications` block supports the following:
+
+* `package_reference_id` - (Required) Specifies the Gallery Application Version resource ID.
+
+-> **NOTE:** The `package_reference_id` should be in the form of `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Compute/galleries/gallery1/applications/application1/versions/version1`.
+
+* `configuration_reference_blob_uri` - (Optional) Specifies the URI to an Azure Blob that will replace the default configuration for the package if provided.
```
Comment (katbyte) [inline website/docs/r/windows_virtual_machine.html.markdown:254]: is reference make sense here?
```suggestion
* `configuration_blob_uri` - (Optional) Specifies the URI to an Azure Blob that will replace the default configuration for the package if provided.
```

File: internal/services/compute/linux_virtual_machine_resource.go [hunk 3]
```diff
@@ -206,6 +206,8 @@ func resourceLinuxVirtualMachine() *pluginsdk.Resource {
 				ValidateFunc: azValidate.ISO8601DurationBetween("PT15M", "PT2H"),
 			},
 
+			"gallery_applications": VirtualMachineGalleryApplicationsSchema(),
```
Comment (katbyte) [inline internal/services/compute/linux_virtual_machine_resource.go:209]: this should be
```suggestion
			"gallery_application": VirtualMachineGalleryApplicationsSchema(),
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! The new property names you suggested totally make sense. Just the feature has just been added to VMSS with these property names (could be found in pr description), shall we use the same one from VMSS here, or use the new name instead, add deprecate those in VMSS?

**Comment by katbyte** [comment]
i think these names are sufficiently better the old ones should be deprecated ?

**Comment by myc2h6o** [comment]
@katbyte got it! I've updated the property names, please take a look.

**Comment by myc2h6o** [comment]
After this change is done, I'll start another pr to update these property names in VMSS resources

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Comment by farwind** [comment]
seems like functionality was released for both linux and windows properly but release notes states only windows vitrual machine twice : 
"azurerm_storage_account - support for the multichannel_enabled property (https://github.com/hashicorp/terraform-provider-azurerm/issues/17999)
azurerm_windows_virtual_machine - support for the gallery_application property (https://github.com/hashicorp/terraform-provider-azurerm/issues/18406)
azurerm_virtual_hub_bgp_connection - support for the virtual_network_connection_id property (https://github.com/hashicorp/terraform-provider-azurerm/issues/18469)
azurerm_windows_virtual_machine - support for the gallery_application property (https://github.com/hashicorp/terraform-provider-azurerm/issues/18406)"

**Comment by katbyte** [comment]
thanks @farwind for pointing that out - i've corrected the changelog

**Pull request merged**